core/src/main/java/hudson/FilePath.java

```diff
@@ -3365,12 +3365,7 @@ private TunneledInterruptedException(InterruptedException cause) {
         private static final long serialVersionUID = 1L;
     }
 
-    private static final Comparator<String> SHORTER_STRING_FIRST = new Comparator<String>() {
-        @Override
-        public int compare(String o1, String o2) {
-            return o1.length()-o2.length();
-        }
-    };
+    private static final Comparator<String> SHORTER_STRING_FIRST = (o1, o2) -> o1.length() - o2.length();
```

Comment:
This can be simplified further using the stock utility method taking `String::length`.

Reply:
```suggestion
    private static final Comparator<String> SHORTER_STRING_FIRST = Comparator.comparingInt(String::length);
```
I imagine Jesse was thinking about something like this :)

Reply:
Something like that, yes.